### DISCUSSION TITLE
SupportAutoInjectedAssets inadvertently changes data type of Response's $original

### DISCUSSION THREAD

**Discussion by legecha** [livewire/livewire#6921] [7. Bugs (archived)]
### Livewire version

v3.0.5

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

1. Create a controller with a method that has an `Illuminate\View\View` return type

```
public function show(Category $category): View
```

2. Write a test that checks the data is valid:

```
    $response = $this->get(route('category.show', $category->slug));

    expect($response)
        ->assertViewHas('category', fn ($category) => $myCategoryToCheck->is($category));
```

3. Run test to receive the following error:

`The response is not a view.`

4a. Check events subscribed to `RequestHandled`:

```
php artisan event:list | grep -A 1 'RequestHandled'
  Illuminate\Foundation\Http\Events\RequestHandled
  ⇂ Closure at: /vendor/livewire/livewire/src/Features/SupportAutoInjectedAssets/SupportAutoInjectedAssets.php:23
```

4b. Check data type before the line that inadvertently overwrites the `$original` variable in the Response (https://github.com/livewire/livewire/blob/3535c5c9f4c4076d66656cedbdc084711054cd2b/src/Features/SupportAutoInjectedAssets/SupportAutoInjectedAssets.php#L33)

```
dump($handled->response->original);
if (str($html)->contains('</html>')) {
    $handled->response->setContent(static::injectAssets($html));
}
dd($handled->response->original);
```

```
Illuminate\View\View {#4737 // vendor/livewire/livewire/src/Features/SupportAutoInjectedAssets/SupportAutoInjectedAssets.php:32
...
""" // vendor/livewire/livewire/src/Features/SupportAutoInjectedAssets/SupportAutoInjectedAssets.php:35
```

4c. Add the following to the test, and it passes:

```
Event::fake([
    RequestHandled::class,
]);
```

### Additional information

I'm sorry if you think this is a Laravel bug/nuance as opposed to a Livewire one. My experience with both isn't good enough to suggest a fix/submit a PR sorry. But basically, it appears that through `SupportAutoInjectedAssets`, the underlying `$original` property on the `Response` object gets overwritten when `setContent()` is called. It looks like `getContent` isn't overwritten, so I don't know whether it's a case of having to use `$handled->response->original` instead and doing some checks there or if there's a better way with the underlying Laravel code.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by nedwors** [comment]
To add to @legecha's notes, I have gone through and reproduced the same issues independently (I've only just discovered this thread)

I have ran debugging with `ray` and determined that this is indeed the cause of changing the type of `$original` from `View` to `string`

To assist with testing, I have created this repository with an applicable failing test: https://github.com/nedwors/livewire-auto-asset-test

I hope this helps! At the moment it has caused breaks with a test suite we have in a project after the upgrade from LW2 to LW3, it would be great to be able to keep the `assertView...` assertions in the tests!
